File: 3704-count-partitions-with-even-sum-difference/count-partitions-with-even-sum-difference.py
class Solution(object):
    def countPartitions(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        s=sum(nums)
        pre=c=0
        for i in range(len(nums)-1):
            pre+=nums[i]
            s-=nums[i]
            if ((pre-s)&1)==0:
                c+=1

        return c

File: 3704-count-partitions-with-even-sum-difference/count-partitions-with-even-sum-difference.java
class Solution {
    public int countPartitions(int[] nums) {
        int sum=Arrays.stream(nums).sum();
        int pre=0,c=0;
        for(int i=0;i<nums.length-1;i++){
            pre+=nums[i];
            sum-=nums[i];
            int res=pre-sum;
            if((res&1)==0) c++;
        }
        return c;
    }
}